### DISCUSSION TITLE
Typescript type errors in Reduce Function

### DISCUSSION THREAD

**Discussion by gold-pudge** [community/community#39383] [Projects and Issues]
### Select Topic Area

Bug

### Body

I have two dummy functions using reduce method. The aim is to count how many characters have of each eye color and 2nd array of possible eye colors.

My point is that typescript gives an error in totalCounterColors on acc[color]:

Element implicitly has an 'any' type because expression of type 'string' can't be used to index type '{}'

and 2nd function totalUniqueColors:

Argument of type 'string' is not assignable to parameter of type 'never'.

I was trying different the type declaration and cast and still can get the problems solved. I am relatively new to typescript and want to understand what's the reasons behind as typescript errors quite cryptic for me. Thanks.

**Comment by DSDmark** [comment]
Hi @yU5295,

it's heard to solve your problem without seeing the code. I just guess why this causing this error.

About your 2nd function, you could try this:-

```js
function totalUniqueColors(): void {
    //....
}
```

About the **never,** I suggest read this. Here is better explanation about that:- [**typeScript Docs**](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#the-never-type)

> if i honest with you typescript is greate no doute about it but Type checking JavaScript introduces more issues than I ever solve in my whole 4-year in computer programming.
